azure/services/scalesets/scalesets.go

```diff
@@ -544,22 +547,33 @@ func (s *Service) getVirtualMachineScaleSetIfDone(ctx context.Context, future *i
 	return converters.SDKToVMSS(vmss, vmssInstances), nil
 }
 
-func (s *Service) generateExtensions() []compute.VirtualMachineScaleSetExtension {
+func (s *Service) generateExtensions() ([]compute.VirtualMachineScaleSetExtension, error) {
 	extensions := make([]compute.VirtualMachineScaleSetExtension, len(s.Scope.VMSSExtensionSpecs()))
 	for i, extensionSpec := range s.Scope.VMSSExtensionSpecs() {
 		extensionSpec := extensionSpec
+		parameters, err := extensionSpec.Parameters(nil)
```

Comment:
@CecileRobertMichon This might be a scope creep but should we also refactor VMSS extensions to be async as well? `VMSSExtensionSpecs()` and `VMExtensionSpecs()` both used to return `ExtensionSpec` which is refactored into `vmextensions.ExtensionSpec`. But currently `vmextensions.ExtensionSpec.Parameters()` creates a `compute.VirtualMachineExtension` while here we want a `compute.VirtualMachineScaleSetExtension`. Another option would be to add a converter function that just copies the fields.

Reply:
yes I think doing both VM and VMSS extensions in the same PR might actually make it easier since they share some code

Reply:
@CecileRobertMichon Cool I've made some changes to refactor VMSS. Could you take a look when you have some time?